practice/intlist.c

```diff
@@ -0,0 +1,172 @@
+// ------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// ------------------------------------------------------------------
+// intlist.c
+// Program for Linked List functions.
+// ------------------------------------------------------------------------------------------------
+#include <stdio.h>
+#include <stdlib.h>
+#include <malloc.h>
+#include "intlist.h"
+// Function to create a new linked list
+LinkedList* Create () {
+	LinkedList* list = (LinkedList*)malloc (sizeof (LinkedList)); // Allocate memory for the list structure
+	if (list == NULL) {
+		printf ("Memory allocation failed\n");
+		return NULL;
+	}
+	list->head = NULL; // Initialize the head pointer to NULL, Indicating an empty list
+	return list;
+}
+
+// Function to add an element to the end of the list
+void Add (LinkedList* list, int val) {
+	Node* newNode = (Node*)malloc (sizeof (Node)); // Allocate memory for the new node
+	if (newNode == NULL) {
+		fprintf (stderr, "Memory allocation failed!\n");
+		exit (EXIT_FAILURE);

```

Comment:
You should not exit the program in case of any error. You should just return the error code. It is the responsibility of the calling function to manage error cases.